config/config.go

```diff
@@ -868,6 +870,20 @@ func ParseConfigString(ctx *ParsingContext, configPath string, configString stri
 //  5. Merge the included config with the parsed config. Note that all the config data is mergeable except for `locals`
 //     blocks, which are only scoped to be available within the defining config.
 func ParseConfig(ctx *ParsingContext, file *hclparse.File, includeFromChild *IncludeConfig) (*TerragruntConfig, error) {
+	if detectInputsCtyUsage(file) {
+		allControls := ctx.TerragruntOptions.StrictControls
+
+		skipDependenciesInputs := allControls.Find(controls.SkipDependenciesInputs)
+		if skipDependenciesInputs == nil {
+			return nil, errors.New("failed to find control " + controls.SkipDependenciesInputs)
+		}
+
+		logger := log.ContextWithLogger(ctx, ctx.TerragruntOptions.Logger)
+		if err := skipDependenciesInputs.Evaluate(logger); err != nil {
```

Comment:
it's a bit confusing, creates the wrong impression that `Evaluate` takes the Logger type. 

```suggestion
		evalCtx := log.ContextWithLogger(ctx, ctx.TerragruntOptions.Logger)
		if err := skipDependenciesInputs.Evaluate(evalCtx); err != nil {
```

Maybe it's even worth to move this code `if detectInputsCtyUsage(file) {....}` into a separate function. Up to you.